### PR TITLE
chore(dagger): Bump CLI version and update dependencies

### PR DESCRIPTION
**Pull request by javirln**
This patch bumps the CLI version of the dagger module and remove unneeded dependencies.

The contract-less support was added when we developed the feature so it's up to date.

Refs #1145

### CODE REVIEW

File: extras/dagger/go.sum [hunk 1]
```diff
@@ -75,8 +75,6 @@ google.golang.org/genproto/googleapis/api v0.0.0-20240520151616-dc85e6b867a5 h1:
 google.golang.org/genproto/googleapis/api v0.0.0-20240520151616-dc85e6b867a5/go.mod h1:RGnPtTG7r4i8sPlNyDeikXF99hMM+hN6QMm4ooG9g2g=
 google.golang.org/genproto/googleapis/rpc v0.0.0-20240515191416-fc5f0ca64291 h1:AgADTJarZTBqgjiUzRgfaBchgYB3/WFTC80GPwsMcRI=
 google.golang.org/genproto/googleapis/rpc v0.0.0-20240515191416-fc5f0ca64291/go.mod h1:EfXuqaE1J41VCDicxHzUDm+8rk+7ZdXzHV0IhO/I6s0=
-google.golang.org/grpc v1.64.0 h1:KH3VH9y/MgNQg1dE7b3XfVK0GsPSIzJwdF617gUSbvY=
```
Comment (migmartri) [inline extras/dagger/go.sum:78]: do you know of why this change? 

did you run any command?

**Review by migmartri — APPROVED** [review]

File: extras/dagger/go.sum (hunk 1 above)
Comment (javirln) [reply extras/dagger/go.sum:78]: Yes, I ran `dagger develop`
